ToolBar checkbox checks, JT Add Button closes and Test Clean up

AlexSCorey opened the pull request:
##### SUMMARY
This addresses #4616  and #4766 .
The Select-All check box in the DataList Toolbar will be checked when the
user clicks on it. Also, the JT Add button closes when the user clicks
elsewhere on the page as well as when the user clicks on the button a second time.
I also added a test and cleaned up some tests in the DataListToolBar file.

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
 - UI

##### AWX VERSION
```
7.0.0
```


##### ADDITIONAL INFORMATION
![JobTemplate](https://user-images.githubusercontent.com/39280967/65432061-8e3ca100-dde8-11e9-8c9a-ee76ab0c0b66.gif)

AlexSCorey committed:
ToolBar checkbox checks and JT Add Button closes and Test Clean up

The Select-All check box in the DataList Toolbar will be checked when the
user clicks on it. Also, the JT Add button closes when the user clicks
elsewhere on the page as well as when the user clicks on the button a second time.
I also cleaned up some tests in the DataListToolBar file.

marshmalien commented on awx/ui_next/src/screens/Template/TemplateList/TemplateList.jsx:
```diff
@@ -215,28 +226,32 @@ class TemplatesList extends Component {
+                        onSelect={this.handleAddSelect}
+                        toggle={<ToolbarAddButton onClick={() => {}} />}
+                        dropdownItems={[
+                          <DropdownItem key="job" isPlain>
```
We can remove isPlain because DropdownItem doesn't accept it as a prop.

AlexSCorey committed:
Removes isPlain prop from DropdownItem

marshmalien commented on awx/ui_next/src/screens/Template/TemplateList/TemplateList.jsx:
```diff
@@ -215,28 +226,32 @@ class TemplatesList extends Component {
+                    <div
+                      ref={node => {
+                        this.node = node;
+                      }}
```
I like that you're using ref to hang onto the DOM node.  We need determine how we plan to keep track of this dropdown node and the [empty list state dropdown](https://github.com/ansible/awx/blob/devel/awx/ui_next/src/screens/Template/TemplateList/TemplateList.jsx#L257). Since the empty list dropdown isn't attached to a ref, the toggle functionality is broken when there are no items in the list. 

My one other concern, `handleAddToggle` will fire on every page click. In the future, we should refactor this so that the event listeners are only added while the dropdown is open.

marshmalien commented on awx/ui_next/src/screens/Template/TemplateList/TemplateList.jsx:
```diff
@@ -215,28 +226,32 @@ class TemplatesList extends Component {
+                        isPlain
+                        isOpen={isAddOpen}
+                        position={DropdownPosition.right}
+                        onSelect={this.handleAddSelect}
```
`this.handleAddSelect` doesn't exist 😧

AlexSCorey replied:
Waaa!?

marshmalien commented on awx/ui_next/src/screens/Template/TemplateList/TemplateList.jsx:
```diff
@@ -215,28 +226,32 @@ class TemplatesList extends Component {
+                        onSelect={this.handleAddSelect}
+                        toggle={<ToolbarAddButton onClick={() => {}} />}
+                        dropdownItems={[
+                          <DropdownItem isHovered={false} key="job">
```
`isHovered` should default to `false` - it looks like there is an open issue in patternfly
https://github.com/patternfly/patternfly-react/issues/2569

AlexSCorey committed:
Removes unused fnc and unnecessary props adds dom node to Empty State

This.node was use for the add button for both empty list and list with data.
This was done to reduce complexity in handleAddToggle() and I don't think it
will cause bugs because those two elements are not rendered at the same time.

marshmalien reviewed (approved):
Awesome job at figuring out how to add the event listener only while the dropdown is open! I tested this out locally, it works great. 🎉

AlexSCorey committed:
fix lint errors
